File: app/src/main/java/com/example/ep2024/di/SecureStorageModule.java
package com.example.ep2024.di;

import android.content.Context;
import com.example.ep2024.config.SecureStorage;
import javax.inject.Singleton;
import dagger.Module;
import dagger.Provides;

@Module
public class SecureStorageModule {

    @Provides
    @Singleton
    SecureStorage provideSecureStorage(Context context) {
        return new SecureStorage(context);
    }

}

File: app/src/main/java/com/example/ep2024/di/modules/SecureStorageModule.kt
package com.example.ep2024.di.modules

import android.content.Context
import com.example.ep2024.config.SecureStorage
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SecureStorageModule {

    @Provides
    @Singleton
    fun provideSecureStorage(context: Context?): SecureStorage {
        return SecureStorage(context)
    }

}